encryption-and-compression-functions: Update for MySQL 8.0 (#16852)
pingcap/docs: functions-and-operators/encryption-and-compression-functions.md

| [`UNCOMPRESSED_LENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_uncompressed-length)                             | Return the length of a string before compression  |
| [`VALIDATE_PASSWORD_STRENGTH()`](https://dev.mysql.com/doc/refman/8.0/en/encryption-functions.html#function_validate-password-strength) | Validate the password strength |

> 

Reviewer: Is this needed?

Author: ```suggestion
```

## Related system variables

The [`block_encryption_mode`](/system-variables.md#block_encryption_mode) variable sets the encryption mode that is used for `AES_ENCRYPT()` and `AES_DECRYPT()`.

The [`validate_password.*`](/system-variables.md) variables influence the `VALIDATE_PASSWORD_STRENGTH()` function.

## Unsupported functions

* `DES_DECRYPT()`, `DES_ENCRYPT()`, `OLD_PASSWORD()`, `ENCRYPT()`: these functions were deprecated in MySQL 5.7 and removed in 8.0.
* Functions only available in MySQL Enterprise [Issue #2632](https://github.com/pingcap/tidb/issues/2632).

## MySQL Compatibility

> * TiDB doesn't support the `STATEMENT_DIGEST()` and `STATEMENT_DIGEST_TEXT()` functions.

Reviewer: Should we create an issue in tidb repository for adding these?

Author: I don't think that's needed. But I'm also not against it...

## Related system variables

The [`block_encryption_mode`](/system-variables.md#block_encryption_mode) variable sets the encryption mode that is used for `AES_ENCRYPT()` and `AES_DECRYPT()`.

> The [`validate_password.*`](/system-variables.md) variables influence the `VALIDATE_PASSWORD_STRENGTH()` function.

Reviewer 2: ```suggestion
The [`validate_password.*`](/system-variables.md) variables affect the `VALIDATE_PASSWORD_STRENGTH()` function.
```

## Unsupported functions

* `DES_DECRYPT()`, `DES_ENCRYPT()`, `OLD_PASSWORD()`, `ENCRYPT()`: these functions were deprecated in MySQL 5.7 and removed in 8.0.
* Functions only available in MySQL Enterprise [Issue #2632](https://github.com/pingcap/tidb/issues/2632).

> ## MySQL Compatibility
> 
> * TiDB doesn't support the `STATEMENT_DIGEST()` and `STATEMENT_DIGEST_TEXT()` functions.

Reviewer 2: ```suggestion
## MySQL compatibility

* TiDB does not support the `STATEMENT_DIGEST()` and `STATEMENT_DIGEST_TEXT()` functions.
```
